diff --git a/core/lib/Drupal/Core/EventSubscriber/FinishResponseSubscriber.php b/core/lib/Drupal/Core/EventSubscriber/FinishResponseSubscriber.php
index b5fc2aaf25..461e64290c 100644
--- a/core/lib/Drupal/Core/EventSubscriber/FinishResponseSubscriber.php
+++ b/core/lib/Drupal/Core/EventSubscriber/FinishResponseSubscriber.php
@@ -221,7 +221,7 @@ protected function isCacheControlCustomized(Response $response) {
    *   A request object.
    */
   protected function setResponseNotCacheable(Response $response, Request $request) {
-    $this->setCacheControlNoCache($response);
+    $this->setCacheControlNoStore($response);
     $this->setExpiresNoCache($response);

     // There is no point in sending along headers necessary for cache
@@ -274,13 +274,25 @@ protected function setResponseCacheable(Response $response, Request $request) {
   }

   /**
-   * Disable caching in the browser and for HTTP/1.1 proxies and clients.
+   * Force the proxy caches and browsers to revalidate the cache before
+   * downloading the request body.  This means a HTTP request occurs each time,
+   * but the browser may skip retrieving the HTTP body if the content is valid.
    *
    * @param \Symfony\Component\HttpFoundation\Response $response
    *   A response object.
    */
   protected function setCacheControlNoCache(Response $response) {
-    $response->headers->set('Cache-Control', 'no-cache, must-revalidate');
+    $response->headers->set('Cache-Control', 'no-cache', TRUE);
+  }
+
+  /**
+   * Disable caching in the browser and for HTTP/1.1 proxies and clients.
+   *
+   * @param \Symfony\Component\HttpFoundation\Response $response
+   *   A response object.
+   */
+  protected function setCacheControlNoStore(Response $response) {
+    $response->headers->set('Cache-Control', 'no-store', TRUE);
   }

   /**
diff --git a/core/modules/image/tests/src/Functional/ImageStylesPathAndUrlTest.php b/core/modules/image/tests/src/Functional/ImageStylesPathAndUrlTest.php
index 6d89a920d8..469c69d458 100644
--- a/core/modules/image/tests/src/Functional/ImageStylesPathAndUrlTest.php
+++ b/core/modules/image/tests/src/Functional/ImageStylesPathAndUrlTest.php
@@ -217,8 +217,8 @@ public function doImageStyleUrlAndPathTests($scheme, $clean_url = TRUE, $extra_s

     if ($scheme == 'private') {
       $this->assertSession()->responseHeaderEquals('Expires', 'Sun, 19 Nov 1978 05:00:00 GMT');
-      // Check that Cache-Control header contains 'no-cache' to prevent caching.
-      $this->assertSession()->responseHeaderContains('Cache-Control', 'no-cache');
+      // Check that Cache-Control header contains 'no-store' to prevent caching.
+      $this->assertSession()->responseHeaderContains('Cache-Control', 'no-store');
       $this->assertSession()->responseHeaderEquals('X-Image-Owned-By', 'image_module_test');

       // Make sure that a second request to the already existing derivative
@@ -263,7 +263,7 @@ public function doImageStyleUrlAndPathTests($scheme, $clean_url = TRUE, $extra_s
     }
     else {
       $this->assertSession()->responseHeaderEquals('Expires', 'Sun, 19 Nov 1978 05:00:00 GMT');
-      $this->assertSession()->responseHeaderNotContains('Cache-Control', 'no-cache');
+      $this->assertSession()->responseHeaderNotContains('Cache-Control', 'no-store');

       if ($clean_url) {
         // Add some extra chars to the token.
